File: client/src/types/request-data.ts
export type RequestDataControlOption = {
  id: number;
  text: string;
};

export type RequestDataControl = {
  name: string;
  type: "text" | "date" | "digit" | "select" | "multiselect" | "textarea";
  label: string;
  value?: string | number | number[];
  required: boolean;
  readonly: boolean;
  options?: RequestDataControlOption[];
};

export type RequestData = {
  title: string;
  controls: RequestDataControl[];
  approverGroups: string[];
  solutionGroups?: string[];
};

File: src/Requests.Domain/Models/RequestStructure/RequestData.cs
namespace Requests.Domain.Models.RequestStructure
{
    public class RequestData
    {
        public string Title { get; set; }

        public RequestDataControl Controls { get; set; }

        public string[] ApproverGroups { get; set; }

        public string[] SolutionGroups { get; set; }
    }
}
